breakpad-symbols/src/http.rs

```diff
@@ -216,11 +216,15 @@ async fn individual_lookup_debug_info_by_code_info(
             if new_url.starts_with('/') {
                 new_url = new_url.strip_prefix('/').unwrap_or(new_url);
             }
-            let mut parts = new_url.split('/');
+
+            // new_url looks like some/path/stuff/xul.pdb/somedebugid/xul.sym and we want the debug
+            // file and debug id portions which are at fixed indexes from the end
+            let mut parts = new_url.rsplit('/');
+            let debug_identifier_part = parts.nth(1)?;
+            let debug_identifier = DebugId::from_str(debug_identifier_part).ok()?;
             let debug_file_part = parts.next()?;
```

Comment:
This won't work as the previous call `parts.nth(1)` doesn't advance the iterator, but you can use `.nth(2)` instead here.

Reply:
I couldn't get it to work, so I looked up `nth` and if I'm reading this correctly, it does advance the iterator:

https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.nth

So then I switched the order of extracting debug_id and debug_file and changed debug_file to `nth(0)`. Then clippy said to change it to `next()`.

Reply:
Oh, I didn't know it advanced the iterator. I check in devdocs documentation and it didn't mention it, maybe it's out-of-date. Anyway, if clippy says so I won't argue :ok_hand: